### PR TITLE
Specs and unit test to avoid forgetting in the future

### PR DESCRIPTION
**Pull request by jannotti**
Pure spec update.

### CODE REVIEW

**Review by tsachiherman — APPROVED** [review]
looks good to me.

File: data/transactions/logic/doc_test.go [hunk 1]
```diff
@@ -77,6 +77,15 @@ func TestOpImmediateNote(t *testing.T) {
 	require.Empty(t, xd)
 }
 
+func TestAllImmediatesDocumented(t *testing.T) {
+	for _, op := range OpSpecs {
+		if len(op.Details.Immediates) > 0 {
+			xd := OpImmediateNote(op.Name)
+			require.NotEmpty(t, xd, "%s has undocumented immediates", op.Name)
```
Comment (jdtzmn) [inline data/transactions/logic/doc_test.go:84]: This is good enough for now, but ideally we'd have a test that confirms the two lengths are equivalent.

**Review by jdtzmn — APPROVED** [review]

**Comment by winder** [comment]
You should put a call to the makefile in data/transactions/logic in `scripts/travis/codegen_verification.sh`, it will automatically fail the build if any changes are made.

**Comment by tsachiherman** [comment]
@jannotti - The changes here are good to go as is. However, if you're planning to follow any of the suggestions here, please let me know and I will hold off with merging this one in.

**Comment by jannotti** [comment]
The code gen check sounds useful. I'll do that tonight.

**Review by algorandskiy — DISMISSED** [review]

File: data/transactions/logic/doc_test.go (hunk 1 above)
Comment (jannotti) [reply data/transactions/logic/doc_test.go]: I added a check that counts braces.  It has to be skipped for a few more complicated opcodes, but it's a nice addition for most.

**Comment by jannotti** [comment]
> You should update `scripts/travis/codegen_verification.sh` with a call to the makefile in data/transactions/logic, it will automatically fail the build if any changes are made.

I did this, does it seem right?
```
echo "Updating TEAL Specs"
make -C data/transactions/logic
```

I gather the idea is that this will be a no-op if the generated files are correctly checked in, so the git checkout remains clean.

**Review by winder — APPROVED** [review]
LGTM

**Pull request merged**